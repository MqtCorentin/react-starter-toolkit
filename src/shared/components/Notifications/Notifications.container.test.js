import React from 'react';
import { render } from '@testing-library/react';
import NotificationsEnhanced from './Notifications.container';

describe('NotificationsEnhanced', () => {
  it('Render <NotificationsEnhanced/>', () => {
    const defaultProps = {
      className: '',
      notifications: [],
      deleteNotification: () => {},
    };
    const { asFragment } = render(<NotificationsEnhanced {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Render <NotificationsEnhanced/> with notifications', () => {
    const defaultProps = {
      className: '',
      notifications: [{ id: 'id', title: 'label' }],
      deleteNotification: () => {},
    };
    const { asFragment } = render(<NotificationsEnhanced {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
