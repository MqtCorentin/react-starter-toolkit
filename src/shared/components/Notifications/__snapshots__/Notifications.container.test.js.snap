// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationsEnhanced Render <NotificationsEnhanced/> 1`] = `
<DocumentFragment>
  <aside
    class=""
  />
</DocumentFragment>
`;

exports[`NotificationsEnhanced Render <NotificationsEnhanced/> with notifications 1`] = `
<DocumentFragment>
  <aside
    class=""
  >
    <div
      class="af-alert af-alert--notification af-alert--error"
      role="alert"
    >
      <button
        class="af-alert__icon-close"
      >
        <i
          class="glyphicon glyphicon-close"
        />
      </button>
      <div
        class="af-alert__title"
      >
        <div
          class="af-alert__title-icon"
        >
          <i
            class="glyphicon glyphicon-minus-sign"
          />
        </div>
        <div
          class="af-alert__title-text"
        >
          label
        </div>
      </div>
    </div>
  </aside>
</DocumentFragment>
`;
